### DISCUSSION TITLE
add entities for dry contact control for G4

### DISCUSSION THREAD

**Discussion by jtrefil** [wills106/homeassistant-solax-modbus#75] [Ideas]
Hi,

is it possible to add entities at G4 to control dry contact that switches surplus via solax adapter box?
![Snímek obrazovky 2022-06-16 081307](https://user-images.githubusercontent.com/5207295/174003715-465e9cf7-f925-4bb2-bbeb-5d37a1728d60.jpg)
[Hybrid X1&X3-G4 ModbusRTU V3.20-English_0324 (2).pdf](https://github.com/wills106/homeassistant-solax-modbus/files/8916168/Hybrid.X1.X3-G4.ModbusRTU.V3.20-English_0324.2.pdf)

And possibly add a contact status monitoring entity, this would help to fine-tune the switching settings.

Thank you.
Jan

**Comment by wills106** [comment]
They are just the registers to read the status. But it looks like the corresponding write registers are on page 50.

I can look at adding them in and a few other changes I have been asked for. But I am a little busy at the min with other projects so I can't give a time frame.

**Reply by jtrefil** [reply]
Great, one day it would be great if you manage to add .. thanks.

**Reply by infradom** [reply]
I mlght also be able to help implement this, but I need to finish another integration/automation task as I subscribed to a dynamic energy prices trial (day ahead price) that starts 1st of july.
@jtrefil : do you have some documentation about this adapter box (just for my personal interest)

**Reply by jtrefil** [reply]
> I mlght also be able to help implement this, but I need to finish another integration/automation task as I subscribed to a dynamic energy prices trial (day ahead price) that starts 1st of july.
> @jtrefil : do you have some documentation about this adapter box (just for my personal interest)

Only this documentation for SolaxAdapterBox, but it can be bypassed by relay.
https://gbc-solino.cz/wp-content/uploads/2021/11/QML_SOLAX_ADAPTER_BOX-EN.pdf
I enclose a work diagram that I created for myself, so I apologize for the unprofessional drawing
[image2022-06-15-041726.pdf](https://github.com/wills106/homeassistant-solax-modbus/files/8936128/image2022-06-15-041726.pdf)
.

**Reply by wills106** [reply]
@jtrefil Is it just the registers and the equivalent write registers from 0x121 - 0x112F that you need or do you also need 0x116 - 0x120 and 0x130 - 0x132 as well?

I have just started to add these in and wondering if you want all of them exposed or just the ones you have highlighted?

**Reply by jtrefil** [reply]
@wills106 the highlighted ones are enough, but as you pointed out earlier these are the read registers for writing are 0x00B6-0x00C3
thank you. 
![Snímek obrazovky 2022-07-04 080259](https://user-images.githubusercontent.com/5207295/177091328-74673b0c-bdeb-4b6f-9565-cd82f6a77074.jpg)

**Comment by wills106** [comment]
@jtrefil Can you please try the Beta Release [0.6.0b1](https://github.com/wills106/homeassistant-solax-modbus/releases/tag/0.6.0b1)

**Reply by jtrefil** [reply]
@wills106 It looks great, thanks a lot. I received the current modbus RTU documentation from solax support, it might be useful
[Hybrid X1&X3-G4 ModbusTCP&RTU V3.21-English_0622 - public version.pdf](https://github.com/wills106/homeassistant-solax-modbus/files/9265951/Hybrid.X1.X3-G4.ModbusTCP.RTU.V3.21-English_0622.-.public.version.pdf)
.

**Comment by wills106** [comment]
Since 0.6.0b5 the button, number & select entities have been shown in the main Integration page.
Do you have any views on where each of the new Dry Contact entities should sit in relation to Control, Configuration & Diagnostic?

It would be helpful if you could feedback so we know where each one best belongs.
I would recommend starting with 0.6.0b9 as I have moved some Generic entities about already.

**Reply by wills106** [reply]
@jtrefil Hi, just wondering if you managed to look into where you think the various entities belong in relation to the DCB?

**Reply by nickcj931** [reply]
- @wills106  - just wanted to see if this was ever implemented? Im looking at possible purchase of the adapter box. If this still needs to be implemented I am happy to help test (I just no nothing about coding this in!!) so if there is appetite happy to assist

**Reply by cleveHEX** [reply]
You can search for the addresses in the solax file, if it misses anything you need, it can be implemented

**Reply by wills106** [reply]
Dry contact is supported, you just need to tick the Dry Contact Box (DCB) option when you setup the Integration.
You can see it in the screenshot https://github.com/wills106/homeassistant-solax-modbus/tree/2023.10.1#installation

I have started to add in the PeakShaving and Generator control, but that's not finished yet.

**Reply by nickcj931** [reply]
AH! couldn't see for looking!... thanks
